### PR TITLE
handling number literals in value sharing

### PR DESCRIPTION
**Pull request by colombod**
image

### CODE REVIEW

**Comment by colombod** [comment]
@shibbas

File: src/polyglot-notebooks-vscode-common/src/utilities.ts [hunk 1]
```diff
@@ -232,6 +232,15 @@ export function parse(text: string): any {
             return Uint8Array.from(buffer.values());
         }
 
+        //handling NaN, Infinity and -Infinity
+        if (value === "Infinity") {
+            return Infinity;
+        } else if (value === "-Infinity") {
+            return -Infinity;
+        } else if (value === "NaN") {
```
Comment (jonsequitur) [inline src/polyglot-notebooks-vscode-common/src/utilities.ts:240]: Micro-optimization micro-nitpick: `NaN` will be by far the most common case here and might should be checked before the others.

File: src/polyglot-notebooks-vscode-common/src/utilities.ts [hunk 2]
```diff
@@ -250,6 +259,15 @@ export function stringify(value: any): string {
...
         }
 
+        //handling NaN, Infinity and -Infinity
+        if (value === Infinity) {
+            return "Infinity";
+        } else if (value === -Infinity) {
+            return "-Infinity";
+        } else if (value !== value) {
+            return "NaN";
+        }
```
Comment (jonsequitur) [inline src/polyglot-notebooks-vscode-common/src/utilities.ts:269]: Any reason to have two copies of this logic?

File: src/polyglot-notebooks/src/connection.ts [hunk 3]
```diff
@@ -303,3 +303,31 @@ export function extractHostAndNomalize(kernelUri: string): string {
...
+        if (value === "Infinity") {
+            return Infinity;
+        } else if (value === "-Infinity") {
+            return -Infinity;
+        } else if (value === "NaN") {
+            return NaN;
+        }
+        return value;
+    });
+}
```
Comment (jonsequitur) [inline src/polyglot-notebooks/src/connection.ts:333]: Can we consolidate these multiple copies of the same logic?

File: src/polyglot-notebooks-vscode-common/src/utilities.ts (hunk 1 above)
Comment (colombod) [reply src/polyglot-notebooks-vscode-common/src/utilities.ts:240]: indeed

Comment (colombod) [reply src/polyglot-notebooks-vscode-common/src/utilities.ts:240]: done

**Pull request merged**